src/Core/Metadata.idr

```diff
@@ -96,7 +98,7 @@ record Metadata where
        currentLHS : Maybe ClosedTerm
        holeLHS : List (Name, ClosedTerm)
        nameLocMap : PosMap (NonEmptyFC, Name)
-       sourcefile : String
+       sourceIdent : Either VirtualIdent (SourceFileType, FileName)
```

Comment:
Why not just an `OriginDesc`? Is it because we want a `FileName` rather than a `ModuleIdent` in `PhysicalIdrSrc`?
Then we could define

```idris
data OriginDesc' : Type -> Type where      
      ||| Anything that originates in physical Idris source files is assigned a
      ||| `PhysicalIdrSrc modIdent`,
      |||   where `modIdent` is the top-level module identifier of that file.
      PhysicalIdrSrc : (ident : a) -> OriginDesc
      ||| Anything parsed from a package file is decorated with `PhysicalPkgSrc fname`,
      |||   where `fname` is path to the package file.
      PhysicalPkgSrc : (fname : FileName) -> OriginDesc
	  Virtual : (ident : VirtualIdent) -> OriginDesc

OriginDesc = OriginDesc' ModuleIdent
MetaOrigin = OriginDesc' FileName

Eq a => Eq (OriginDesc' a) where (...)
Functor OriginDesc' where (...)
```

and `compareOrigins` becomes `(pathToNS' wdir sdir) <$> finfo == ident`

Reply:
Yes, good idea. Let me do that